std/typecons.d

```diff
@@ -804,27 +804,33 @@ if (distinctFieldNames!(Specs))
          *   $(LI A positive integer if the expression `v1 > v2` is true.)
          *   $(LI 0 if the expression `v1 == v2` is true.))
          */
-        int opCmp(R)(R rhs)
+        auto opCmp(R)(R rhs)
         if (areCompatibleTuples!(typeof(this), R, "<"))
         {
             static foreach (i; 0 .. Types.length)
             {
                 if (field[i] != rhs.field[i])
                 {
+                    static if (isFloatingPoint!(Specs[i]) || isFloatingPoint!(typeof(rhs.field[i])))
```

Comment:
`Specs` includes the optional field names but `Types` should work.
```suggestion
                    static if (isFloatingPoint!(Types[i]) || isFloatingPoint!(typeof(rhs.field[i])))
```

Reply:
I might also be a good idea to split this up, e.g. when comparing `Tuple!(int)` and `Tuple!(double)`